src/main/java/baritone/pathing/movement/Movement.java

```diff
@@ -108,32 +111,32 @@ public double calculateCostWithoutCaching() {
      */
     @Override
     public MovementStatus update() {
-        player().capabilities.isFlying = false;
+        ctx.player().capabilities.isFlying = false;
         currentState = updateState(currentState);
-        if (MovementHelper.isLiquid(playerFeet())) {
+        if (MovementHelper.isLiquid(ctx, ctx.playerFeet())) {
             currentState.setInput(Input.JUMP, true);
         }
-        if (player().isEntityInsideOpaqueBlock()) {
+        if (ctx.player().isEntityInsideOpaqueBlock()) {
             currentState.setInput(Input.CLICK_LEFT, true);
         }
 
         // If the movement target has to force the new rotations, or we aren't using silent move, then force the rotations
         currentState.getTarget().getRotation().ifPresent(rotation ->
-                Baritone.INSTANCE.getLookBehavior().updateTarget(
+                baritone.getLookBehavior().updateTarget(
```

Comment:
Remind me why baritone can't just be a parameter passed to update()??

Reply:
There would be a lot of other methods to add a parameter to and Movement is Baritone-instance based anyways

Reply:
Like what? In both scenarios, the code within updateState has a reference to the Baritone?

Reply:
fineeeeeeee